### DISCUSSION TITLE
Pretty sure 3rd party font blocking doesn't work

### DISCUSSION THREAD

**Discussion by sheerun** [uBlockOrigin/uAssets#14954] [General]
### Prerequisites

- [X] This is not a [support issue or a question](https://old.reddit.com/r/uBlockOrigin/)
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uAssets/issues) to avoid opening a duplicate issue
- [X] I am running the latest version of uBlock Origin

### I tried to reproduce the issue when...

- [X] uBO is the only extension
- [X] uBO with default lists/settings
- [X] using a new, unmodified browser profile

### URL(s) where the issue occurs

`https://www.logitech.com/pl-pl/products/mice.html`

### Describe the issue

If I add `*$font,third-party` then if I open Network tab of Developer Tools, I see that all first-party fonts are blocked, for example: `https://www.logitech.com/asset/fonts/lineto-brown-pro-bold.woff`.

### Screenshot(s)

<img width="1252" alt="Screenshot 2022-09-20 at 18 42 46" src="https://user-images.githubusercontent.com/292365/191316358-4008dc35-3914-48d3-b66a-7d13922ad465.png">

### uBlock Origin version

1.44.2

### Browser name and version

Firefox 104.0.2 

### Settings

Add `*$font,third-party` to My Filters

**Comment by mapx-** [comment]
It's about cname
https://www.reddit.com/r/uBlockOrigin/comments/f8qnpc/ublock_origin_1250_with_cname_uncloaking_is_out/

For example the first-party 
`https://www.logitech.com/etc.clientlibs/logitech/clientlibs/replatform/resources/fonts/logi-icon-2.woff2`

is uncloaked as third party `https://d1hy6rfccmvxt8.cloudfront.net/` which your third-party filter will block and so in the Network tab you'll see uBO is blocking ... first-party (which instead is third-party)

You can open the logger and see the cnames
![image](https://user-images.githubusercontent.com/10303732/191321001-ce17cf20-711c-4a36-a4f2-e4eea1d1fbea.png)

**Comment by mapx-** [comment]
if you add (nullifying the cname operation)
`@@||logitech.com^$cname`

You'll get a "clean" network tab

**Comment by sheerun** [comment]
Is there a way to universally block 3rd party <strike>cookies</strike> fonts? i.e. without something like `logitech.com` in rules

**Reply by MasterKia** [reply]
Why not block third-party cookies in browser settings?

**Reply by sheerun** [reply]
Sorry, I meant fonts, not cookies

**Reply by gorhill** [reply]
The logitech filter is to bypass uncloaking, so if you disable uncloaking in uBO settings, you won't need such filters. The consequence though is to lose benefits of uncloaking. It's a choice you have to make.

**Reply by sheerun** [reply]
Could I both have uncloaking enabled and prevent 3rd party fonts from loading, without website-specific filters? I want both and I think most people would like it this way :)

**Reply by gorhill** [reply]
> prevent 3rd party fonts from loading

`d1hy6rfccmvxt8.cloudfront.net` **is** 3rd-party, uBO merely made this information visible, so the behavior you want occurred. It's the webmaster who decided to make those 3rd-party fonts appear as 1st-party, so if your argument is that the webmaster should get to decide what is getting blocked by uBO, this goes against **your** wish to block 3rd-party fonts.

**Reply by sheerun** [reply]
But html website `www.logitech.com` is itself cnamed to `d1hy6rfccmvxt8.cloudfront.net`, so any fonts loaded from `d1hy6rfccmvxt8.cloudfront.net` should be considered first party? In other words anything loaded from `d1hy6rfccmvxt8.cloudfront.net` should be considered 1st party, not only fonts

**Reply by sheerun** [reply]
Additional note: I get this issue on most websites, as most websites use CDN and static hosting for their primary domain. I guess the issue should say that 3rd party font blocking works too well :)

**Reply by gorhill** [reply]
> But html website `www.logitech.com` is itself cnamed to `d1hy6rfccmvxt8.cloudfront.net`

Ok I see, then it's <https://github.com/uBlockOrigin/uBlock-issues/issues/1062>, I still need to work on this.
